package.json

```diff
@@ -34,9 +34,9 @@
     "docs:preview": "vitepress preview"
   },
   "engines": {
-    "node": ">=20.9"
+    "node": ">=20.9",
+    "npm": "^10.0.0"
```

Comment:
Maybe I misunderstood this from your PR description:

> That’s why currently, in CI, the installed npm version (10.1.0) does not match the pkg.engines.npm value (arguably, actions/setup-node should use that).

If in CI the installed npm version is 10.1.10 then this matches the `pkg.engines.npm` value doesn't it?

Reply:
> If in CI the installed npm version is 10.1.10 then this matches the `pkg.engines.npm` value doesn't it?

No, it does not. It _happens_ to fulfill it because of the Node version we’re installing (which also installs npm), but there is no guarantee for that and there exists no mechanism in `actions/setup-node` today that would allow us to configure the npm version.

If we want to make sure the npm version we define in `pkg.engines.npm` is used in CI, we would need to manually install it after every instance of `actions/setup-node` in CI. Considering this, it turns out the need to manually install pnpm in CI is actually an advantage.

Reply:
Oh I think I get you.

From what I remember `npm` versions are bundled with `node` aren't they? So depending on which version of `node` you have pin to you get a certain version of `npm` also?

Reply:
> From what I remember `npm` versions are bundled with `node` aren't they?

Yes. That’s the reason why it’s _that_ particular version of npm that is installed in CI (as opposed to the one already available in GitHub Actions by default).

> So depending on which version of `node` you have pin to you get a certain version of `npm` also?

Correct. Any particular version of node does come with a particular version of npm (presumably because that’s the latest stable version npm that was available at the time that node version was released). There’s no reason however to not select/install a completely different npm version so long as it works with that node version (where there should almost never be any version conflicts).

Reply:
Ok so I think what we are saying here is that as we pin `node` that also pins the version of `npm`, in which case we could just remove this entry from `pkg.engines.npm`? 

I'm just thinking of less stuff to think about when we come to do node upgrades.

Not saying we need to do it here, just asking more than anything.

Reply:
Oh I thought you said we weren't controlling it?

> but there is no guarantee for that and there exists no mechanism in actions/setup-node today that would allow us to configure the npm version.

Or perhaps better said I'd understood we are controlling it via `node` pinning.

Anyway doesn't matter I was half thinking out loud. I think this PR is good 1 sec lemme approve.

Reply:
> Oh I thought you said we weren't controlling it?

This is specific to CI where one way or another, the tooling we use (actions/setup-node) doesn’t honor `pkg.engines.npm` and always installs its own npm version.

Locally, that’s not the case and `pkg.engines.npm` gives us the useful validation for whether a developer uses a compatible version.